### PR TITLE
literals: special-case +-Infinity, undefined, and NaN

### PR DESCRIPTION
**Pull request by zkat**
Fixes: #76

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -496,6 +496,9 @@ LiteralMatchBinding :
   NumericLiteral
   StringLiteral
   RegularExpressionLiteral
+  StringNumbericLiteral
```
Comment (ljharb) [inline README.md]: `Numeric`

This seems like it adds a lot of things beyond just ±Infinity. Is that the intention?

Comment (zkat) [reply README.md]: https://tc39.github.io/ecma262/#prod-StrNumericLiteral

Afaict, this doesn't add anything other than Infinity and some whitespace.

If anything, I could switch this to be `StrNumericLiteral` instead, just to remove the pointless whitespace expressions.

Comment (ljharb) [reply README.md]: I might be reading it wrong - spec grammar is far from my strong suit - but https://tc39.github.io/ecma262/#prod-StrNumericLiteral seems to suggest it includes all number literals, and that if you want to include only Infinity, you’d want to type a literal “Infinity” [into the grammar](https://tc39.github.io/ecma262/#prod-StrUnsignedDecimalLiteral)

Comment (zkat) [reply README.md]: Yep, I think this change means we could replace `NumericLiteral` with `StrNumericLiteral altogether.

Comment (ljharb) [reply README.md]: ah, i missed that you already allow numeric literals. Makes sense now :-)

**Pull request closed without merging**